sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UseWhileLoopInstead.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class Program
+    {
+        public Program()
+        {
+            for (int l = 0; l < 10; l++)
+            {
+            }
+
+            for (char c = 'a'; c <= 'z'; c++)
+            {
+            }
+
+            int i;
+            int j = 10;
+            for (i = 0, Console.WriteLine("Start: {0}", i); i < j; i++, j--, Console.WriteLine("i={0}, j={1}", i, j))
+            {
+                // Body of the loop.
+            }
+
+            var x = 5;
+            for (; x < 10;) // Noncompliant {{Replace this 'for' loop with a 'while' loop.}}
+//          ^^^
+            {
+            }
+
+            for (; ; ) // Noncompliant
+            {
+
+            }

```

Comment:
Other compliant cases:
* just the declaration is null
* just the incrementor is null